### DISCUSSION TITLE
TP 3 - 1.2 Impresora de arreglos

### DISCUSSION THREAD

**Discussion by SantiagoBravo05** [INGCOM-UNRN-P1/cursada-2023#98] [Preguntas y respuestas] [answered]
Hola, estoy tratando de hacer el tp 3 y no entiendo bien como imprimir el arreglo. Segun lo que investigue hay que imprimir caracter por caracter del arreglo, no hay forma de imprimirlo de una?
Por ejemplo esto no funciona:
```C
    printf("Arreglo resultante: ");
    printf ("%c", arreglo[capacidad]);
```
Pero esto si:
```C
    printf("Arreglo resultante: ");
    while (i < capacidad)
    {
        printf("%c ", arreglo[i]);
        i=i+1;
    }
```
No entiendo por qué y si siempre que se imprima un arreglo hay que hacerlo carácter por carácter.

**Comment by martinvilu** [comment, marked as answer]
> hay que imprimir caracter por caracter del arreglo

Sí

```C
printf ("%c", arreglo[capacidad]);
```
Esto indica un solo lugar del arreglo (la posición indicada por la variable `capacidad`)
Y el especificador `%c` es para caracteres individuales.

**Reply by SantiagoBravo05** [reply]
Si usara un especificador para cadenas tampoco podría?

**Reply by martinvilu** [reply]
¿Es un arreglo o es una cadena?

Como comentaron en clases, "lo que pasaba en Python, quedo en Python"
